test/D07/TC01.cs

```diff
@@ -0,0 +1,82 @@
+﻿using aoc.D07;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Collections.Generic;
+
+namespace aoc.test.D07
+{
+	[TestClass]
```

Comment:
I would use a data driven approach since you have a lot of duplication here.

```csharp
	[TestClass]
	public class TC01
	{
		[TestMethod]
		[DataRow("./D07/ex1.txt", 4)]
		[DataRow("./D07/input.txt", 226)]
		public void PartI(string filepath, int expected)
		{
			List<Bag> bags = Parser.Parse(filepath);
			int actual = 0;

			foreach (var bag in bags)
				if (bag.CanHoldShinyGoldBag(bags))
					actual++;

			Assert.AreEqual(expected, actual);
		}

		[TestMethod]
		[DataRow("./D07/ex1.txt", 32)]
		[DataRow("./D07/ex2.txt", 126)]
		[DataRow("./D07/Input.txt", 9569)]
		public void Part2(string filepath, int expected)
		{
			List<Bag> bags = Parser.Parse(filepath);

			var bag = bags.Find(b => b.Color == "shiny gold");

			var actual = bag.TotalBagsNeeded(bags) - 1;  //excluding itself

			Assert.AreEqual(expected, actual);
		}
	}
```

Reply:
super! done